File: TicTacToe/TicTacToe/Player.vb
Public Class Player
    Implements IPlay

    Private _name As String
    Private _hand As HandType
    Private _win As Boolean
    Private _cpu As Boolean

    Public Sub New(ByVal type As UserType)
        Select Case type
            Case UserType.User
                CreateUserData()
            Case UserType.CPU
                CreateCpuData()
        End Select
    End Sub

    Private Sub CreateCpuData()
        _name = "CPU"
        _cpu = True
        _hand = HandType.Cross
    End Sub

    Private Sub CreateUserData()
        Console.Write("input your name: ")
        _name = Console.ReadLine()

        If _name.Length = 0 Then
            Console.WriteLine("input error")
            Console.WriteLine("your name is nothing.")
            _name = "nothing"
        End If
        _hand = HandType.Circle

    End Sub

    Public ReadOnly Property IsCPU As Boolean
        Get
            Return _cpu
        End Get
    End Property

    Public ReadOnly Property IsWin As Boolean
        Get
            Return _win
        End Get
    End Property

    Public Sub Play(ByRef gameboard As GameBoard) Implements IPlay.Play
        If _cpu = True Then
            Dim rnd As New Random()
            Dim rndvalue As Integer

            rndvalue = rnd.Next(1, 10)

            While (gameboard.IsCellEmpty(rndvalue) = False)
                rndvalue = rnd.Next(1, 10)
            End While

            gameboard.UpdateCell(rndvalue, _hand)
        Else
            gameboard.SelectShowDisplay()
            gameboard.UpdateCell(GetSelectCellNumber(gameboard), _hand)
        End If
    End Sub

    Private Function GetSelectCellNumber(ByRef gameboard As GameBoard) As Integer
        Dim num As Integer
        Dim str As String

        While True
            Console.Write("input put area number. :")
            str = Console.ReadLine()

            If IsNumeric(str) = False Then
                Console.WriteLine("error, retry.")
                Continue While
            End If

            num = Int32.Parse(str)

            If (num < 1 Or num > 9) Then
                Console.WriteLine("error, retry.")
                Continue While
            End If

            If gameboard.IsCellEmpty(num) = False Then
                Console.WriteLine("not empty, retry input area number.")
                Continue While
            Else
                Exit While
            End If

        End While

        Return num
    End Function

    Public Sub WinnerInfo() Implements IPlay.WinnerInfo
        Console.WriteLine("winner is {0}.", _name)
    End Sub

    Public Function GetHand() As HandType Implements IPlay.GetHand
        Return _hand
    End Function
End Class

File: TicTacToe/TicTacToe/Player.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace TicTacToe
{
	class Player : IPlay
	{
        private string _name;
        private HandType _hand;
        private PlayerType _type;
        private IPlayLogic _logic;

        public Player(PlayerType type)
        {
            this._type = type;
            switch (type)
            {
                case PlayerType.CPU:
                    _name = "CPU";
                    _hand = HandType.Cross;
                    _logic = SelectLogic();
                    break;
                case PlayerType.USER:
                    _name = InputYourName();
                    _hand = HandType.Circle;
                    break;
            }
        }

        #region IPlay member

        public void Play(GameBoard board)
        {
            board.DisplayForSelectNumber();
            int result = 0;
            if (_type == PlayerType.CPU)
            {
                result = getNumberForLogic(board);
            }
            else
            {
                result = normalSelect(board);
            }

            board.UpdateCell(board.GetCell(result), (int)_hand);
            board.Display();
        }

        public void WinInfo()
        {
            Console.WriteLine("{0} is winner.", _name);
            if (_type == PlayerType.CPU)
            {
                Console.WriteLine("you are lose....  λ....");
                Console.WriteLine("Kuyashiidesu!!!");
            }
        }

        public int GetHand()
        {
            return (int)_hand;
        }

        #endregion

#region private methods
        private IPlayLogic SelectLogic()
        {
            PlugManager manager = new PlugManager();

            int i = 0;
            foreach (IPlayLogic p in manager.getPlugList())
            {
                Console.WriteLine("{0}: {1}", i, p.GetName());
                i++;
            }

            if (manager.getPlugList().Count == 1)
            {
                return manager.getPlugList()[0];
            }

            while (true)
            {
                Console.Write("select cpu logic: ");
                string pluginnum = Console.ReadLine();
                int logicnum = 0;
                if (Int32.TryParse(pluginnum, out logicnum))
                {
                    List<IPlayLogic> list = manager.getPlugList();
                    Thread.Sleep(1000);
                    if (list.Count > logicnum)
                    {
                        Console.WriteLine("ok load plugin {0}", list[logicnum].GetName());
                        return list[logicnum];
                    }
                    else
                    {
                        continue;
                    }

                }
                else
                {
                    continue;
                }
            }
        }

        private string InputYourName()
        {
            Console.Write("input player name: ");


            string name = Console.ReadLine();

            while (name.Length == 0)
            {
                Console.WriteLine("error, retry input");
                Console.Write("input player name: ");
                name = Console.ReadLine();
            }

            return name;
        }

        private bool IsGameNumeric(int num)
        {
            return 0 < num && num < 10;
        }

        private int getNumberForLogic(GameBoard board)
        {
            int result = _logic.GetNumber((IBoard)board);

            while (IsGameNumeric(result) == false ||
                board.IsCellEmpty(result) == false)
            {
                result = _logic.GetNumber((IBoard)board);
            }
            return result;
        }

        private int normalSelect(GameBoard board)
        {
            return InputYourNumber(board);
        }

        private int InputYourNumber(GameBoard board)
        {
            int num = 0;

            while (true)
            {
                Console.Write("input area number.: ");
                string numstr = Console.ReadLine();
                if (Int32.TryParse(numstr, out num) == false)
                {
                    Console.WriteLine("error input. plz retry...");
                    continue;
                }

                if (num < 1 || 9 < num)
                {
                    Console.WriteLine("error number. plz retry.");
                    continue;
                }

                if (board.IsCellEmpty(num) == false)
                {
                    Console.WriteLine("sorry, not empty. plz retry...");
                    continue;
                }

                break;
            }

            return num;
        }
#endregion
    }
}
